### PR TITLE
[REVERT] Revert "[SMALLFIX] Reorder common and client in alluxio-core-server"

### PR DESCRIPTION
**Pull request by peisun1115**
Reverts Alluxio/alluxio#4534

### CODE REVIEW

File: core/server/pom.xml [hunk 1]
```diff
@@ -35,19 +35,14 @@
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
     </dependency>
-    <!-- alluxio-core-common should come before alluxio-core-client as alluxio-core-client shade some dependencies
-    such as protobuf. We want to use the unshaded ones in alluxio-core-server.
-    TODO(peis, jiri): Do not depend on alluxio-core-client in alluxio-core-server to avoid depending on
```
Comment (jsimsa) [inline core/server/pom.xml]: can we keep the TODO ;-)

Comment (jsimsa) [inline core/server/pom.xml]: `TODO(jiri): Do not depend on alluxio-core-client in alluxio-core-server.`

Comment (peisun1115) [inline core/server/pom.xml]: This PR is auto-generated. I can close this PR and open another PR

Comment (jsimsa) [inline core/server/pom.xml]: you can also edit the PR through github UI for such simple changes

Comment (peisun1115) [inline core/server/pom.xml]: good idea. done

**Pull request merged**